packages/flutter/lib/src/material/navigation_bar_theme.dart

```diff
@@ -0,0 +1,211 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'navigation_bar.dart';
+import 'theme.dart';
+
+/// Defines default property values for descendant [NavigationBar]
+/// widgets.
+///
+/// Descendant widgets obtain the current [NavigationBarThemeData] object
+/// using `NavigationBarTheme.of(context)`. Instances of
+/// [NavigationBarThemeData] can be customized with
+/// [NavigationBarThemeData.copyWith].
+///
+/// Typically a [NavigationBarThemeData] is specified as part of the
+/// overall [Theme] with [ThemeData.navigationBarTheme].
+///
+/// All [NavigationBarThemeData] properties are `null` by default.
+/// When null, the [NavigationBar] will use the values from [ThemeData]
+/// if they exist, otherwise it will provide its own defaults based on the
+/// overall [Theme]'s textTheme and colorScheme. See the individual
+/// [NavigationBar] properties for details.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+@immutable
+class NavigationBarThemeData with Diagnosticable {
+  /// Creates a theme that can be used for [ThemeData.NavigationBarTheme].
+  const NavigationBarThemeData({
+    this.height,
+    this.backgroundColor,
+    this.indicatorColor,
+    this.unselectedLabelTextStyle,
+    this.selectedLabelTextStyle,
+    this.unselectedIconTheme,
+    this.selectedIconTheme,
+    this.labelBehavior,
```

Comment:
nit: We should open a tracking bug for nav rail alignment

in Nav Rail, we will likely want to replace labelType with labelBehavior, and align the nomenclature of the enum names.

Also, we may want to add indicatorColor to Nav Rail.

Finally, we may want to expose more customizations to the indicator, such as shape, or direct canvas painting.

Reply:
Is elevation intentionally left out?

Reply:
About elevation, yes, because new spec says it shouldn't be elevated.

Reply:
Since the spec will specify no drop shadow, I intentionally left elevation out. I think it _could_ be added back later if folks want it but a simpler API to start with is preferred